lib/Doctrine/ORM/Query/Lexer.php

```diff
@@ -164,7 +164,7 @@ protected function getType(&$value)
                 return self::T_STRING;
 
             // Recognize identifiers
-            case (ctype_alpha($value[0]) || $value[0] === '_'):
+            case (ctype_alpha($value[0]) || $value[0] === '_' || $value[0] === '\\'):
```

Comment:
And also have a look [here](https://github.com/doctrine/doctrine2/blob/master/tests/Doctrine/Tests/ORM/Query/SelectSqlGenerationTest.php#L465) where a FQCN with a leading `\` is used in INSTANCE OF.

In the BCNF, both `InstanceOfParameter` and `RangeVariableDeclaration` build on `AbstractSchemaName`, which in turn is an `identifier`. `identifier` itself is a little... underspecified?